hls4ml/backends/vivado/passes/fifo_depth_optimization.py

```diff
@@ -0,0 +1,102 @@
+        # profiled
+
+        if init_large_fifo:
+
+            for k, v in model.output_vars.items():
+                if model.config.get_config_value('Backend') == 'Vivado' and (v == model.get_input_variables()[0] or v == model.get_output_variables()[0]):
+                    continue
+                v.pragma = (v.pragma[0], 100000)
+
+        model.write()
+        model.build(reset=False, csim=True, synth=True, cosim=True, validation=False, export=False, vsynth=False)
+
+        with open(
+                model.config.get_output_dir() + '/' + model.config.get_project_name() + '_prj' + '/solution1/sim/verilog/fifo_opt.vcd') as vcd_file:
+            vcd = VcdParser()
+            vcd.parse(vcd_file)
+            data = vcd.scope.toJson()
+
+        # wrapper fifos - useful only with VivadoAccelerator backend
+        if model.config.get_config_value('Backend') == 'VivadoAccelerator':
+            for i in range(1, len(data['children'][0]['children'][0]['children'])):
+                # wrapper fifos
+                self._populate_values(data['children'][0]['children'][0]['children'][i]['name'],
+                                      data['children'][0]['children'][0]['children'][i]['children'][0]['data'],
+                                      data['children'][0]['children'][0]['children'][i]['children'][1]['data'])
+
+            n_elem = len(data['children'][0]['children'][0]['children'][0]['children'])
+            for i in range(n_elem):
+                name   = data['children'][0]['children'][0]['children'][0]['children'][i]['name']
+                data_p = data['children'][0]['children'][0]['children'][0]['children'][i]['children'][0]['data']
+                depth  = data['children'][0]['children'][0]['children'][0]['children'][i]['children'][1]['data']
+                self._populate_values(name, data_p, depth)
+        else:
+            n_elem = len(data['children'][0]['children'][0]['children'])
+            for i in range(n_elem):
+                name = data['children'][0]['children'][0]['children'][i]['name']
+                data_p = data['children'][0]['children'][0]['children'][i]['children'][0]['data']
+                depth = data['children'][0]['children'][0]['children'][i]['children'][1]['data']
+                self._populate_values(name, data_p, depth)
+
+        maxs = [{'name': i['name'], 'max': i['max'], 'depth': i['depth']} for i in self.values]
+
+        with open(model.config.get_output_dir() + '/max_depth.json', 'w') as f:
+            json.dump(maxs, f, indent=4)
+
+        for k, v in model.output_vars.items():
+            filtered_max = [x['max'] for x in maxs if v.cppname in x['name']]
+            if len(filtered_max) == 0:
+                continue
+            if len(filtered_max) > 1:
+                print('WARNING! Check names of FIFOs')
+            v.pragma = (v.pragma[0], filtered_max[0] + 1)
+
+        # will not work with multiple i/o variables
```

Comment:
There should be an assert for this

Reply:
There is one in `vivado_accelerator_config.py`